addOns/automation/src/main/java/org/zaproxy/addon/automation/jobs/DelayJob.java

```diff
@@ -46,6 +46,7 @@ public DelayJob() {
 
     @Override
     public void verifyParameters(AutomationProgress progress) {
+        applyParameters(progress);
```

Comment:
It's preferable to move `JobUtils.applyParamsToObject` here than call `applyParameters`, they should not be applied until the job is actually run.

Reply:
Do you mean instead of calling `applyParameters(progress);`
use `JobUtils.applyParamsToObject` to apply it on a local variable instead of applying it to the member `this.parameters`?

```
Parameters tempParameters = new Parameters();
JobUtils.applyParamsToObject(
                (LinkedHashMap<?, ?>) jobData.get("parameters"),
                tempParameters,
                this.getName(),
                null,
                progress);
                
  String timeStr = tempParameters.getTime();
        try {
            new HhMmSs(timeStr);
        } catch (Exception e) {
            progress.error(Constant.messages.getString("automation.error.delay.badtime", timeStr));
        }
```

Reply:
Clarified via IRC:

Move `JobUtils.applyParamsToObject` logic to verify and remove code from `applyParameters`. So `applyParameters` is empty and `verifyParameters` will execute `JobUtils.applyParamsToObject` + validation